packages/main/src/MessageView.ts

```diff
@@ -0,0 +1,246 @@
+import customElement from "@ui5/webcomponents-base/dist/decorators/customElement.js";
+import property from "@ui5/webcomponents-base/dist/decorators/property.js";
+import slot from "@ui5/webcomponents-base/dist/decorators/slot.js";
+import event from "@ui5/webcomponents-base/dist/decorators/event.js";
+import litRender from "@ui5/webcomponents-base/dist/renderer/LitRenderer.js";
+import Integer from "@ui5/webcomponents-base/dist/types/Integer.js";
+import type I18nBundle from "@ui5/webcomponents-base/dist/i18nBundle.js";
+import { getI18nBundle } from "@ui5/webcomponents-base/dist/i18nBundle.js";
+import SegmentedButton, { SegmentedButtonSelectionChangeEventDetail } from "./SegmentedButton.js";
+import MessageViewItem from "./MessageViewItem.js";
+import ListItemType from "./types/ListItemType.js";
+
+import MessageViewTemplate from "./generated/templates/MessageViewTemplate.lit.js";
+
+// Styles
+import MessageViewCss from "./generated/themes/MessageView.css.js";
+
+// Texts
+import {
+	MESSAGE_VIEW_MORE_INFORMATION,
+} from "./generated/i18n/i18n-defaults.js";
+import List from "./List.js";
+import Icon from "./Icon.js";
+
+enum MessageType {
+	Error = "Error",
+	Warning = "Warning",
+	Success = "Success",
+	Info = "Info"
+}
+
+enum MessageTypeIcon {
+	Info = "information",
+	Success = "status-positive",
+	Error = "error",
+	Warning = "status-critical"
+}
+
+enum DesignClassesMapping {
+	Info = "information",
+	Success = "success",
+	Error = "error",
+	Warning = "warning",
+}
+
+enum MessageViewMode {
+	// eslint-disable-next-line @typescript-eslint/no-shadow
+	List = "list",
+	Details = "details",
+}
+
+type MessageItem = {
+	message: MessageViewItem,
+	messageIcon: string,
+	messageDesign: string,
+	position: number,
+	visible: boolean,
+}
+
+/**
+ * @class
+ *
+ * <h3 class="comment-api-title">Overview</h3>
+ *
+ * The <code>ui5-message-view</code> component is used to display a summarized list of different types of messages (error, warning, success, and information messages).
+ *
+ * <h3>Usage</h3>
+ *
+ * You can use the message view to display messages that are not related to form or table fields.
+ * These messages are triggered in response to a user action.
+ * Although the message view can be embedded within various controls, we recommend that you use it only within a dialog.
+ * Use the message view if you want to display multiple messages triggered by an action within a disruptive dialog.
+ *
+ * For the <code>message-view</code>
+ * <h3>ES6 Module Import</h3>
+ *
+ * <code>import @ui5/webcomponents/dist/MessageView.js";</code>
+ *
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webc.main.MessageView
+ * @extends sap.ui.webc.base.UI5Element
+ * @tagname message-view
+ * @public
+ */
+@customElement({
+	tag: "ui5-message-view",
+	renderer: litRender,
+	themeAware: true,
+	languageAware: true,
+	styles: MessageViewCss,
+	template: MessageViewTemplate,
+	dependencies: [SegmentedButton, List, Icon, MessageViewItem],
```

Comment:
In dependencies we described everything that we need to make the component to work correctly (for example if we use something in the template). Please add CustomListitem. Also MessageViewItem is going to be used from the application developers so if they want to use it they have to import it so we can remove it.

Reply:
Added SegmentedButtonItem, CustomListItem and Link as dependencies. Removed MessageViewItem dependency